sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/OperationStatusResult.Serialization.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+
+namespace Azure.ResourceManager.Models
+{
+    public partial class OperationStatusResult : IUtf8JsonSerializable, IJsonModel<OperationStatusResult>
+    {
+        /// <param name="writer"> The JSON writer. </param>
+        /// <param name="options"> The client options for reading and writing models. </param>
+        protected virtual void JsonModelWriteCore(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        {
+            var format = options.Format == "W" ? ((IPersistableModel<OperationStatusResult>)this).GetFormatFromOptions(options) : options.Format;
+            if (format != "J")
+            {
+                throw new FormatException($"The model {nameof(OperationStatusResult)} does not support writing '{format}' format.");
+            }
```

Comment:
I think the format validation is not part of the writecore method.
The interface implementation method should validate the format.

Reply:
OK MGC did this: https://github.com/microsoft/typespec/blob/cea699f73bcc4ff0f4618841dcac010d44bbf5ed/packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/Models/ProjectedModel.Serialization.cs#L32-L36
I think this is incorrect. I will talk with Michael to see how he thinks about this.

Reply:
now the implementation here is inconsistent.
If we want to validate the format in the write core, we should do it for all the models including the other two ResourceData and TrackResourceData.
Or none of them we do the validation here.
Please align